### ISSUE TITLE
Avoiding Running outside of field on corners

### ISSUE THREAD

**Issue by coatsy35** [Courseplay/Courseplay_FS22#2758]
I've been thinking a lot in the last week about implements and how they handle sharp corners and the tractor going outside of the field boundary. Quite often IRL you would not start your headland run on the blue dot in my picture below, you would back up to the hedge/ditch and lower your implement and start the path further back against the hedge/ditch where the blue dashed lines shows, this has the affect's that the next pass around the headland, you do not have to pull all the way forward towards the hedge, you can stop short and the next path will complete the work. Would something like this work in CP?

Picture below, hard to translate into a drawing. Basically the tractor starts further back on the dashed line rather than at the blue circle/dot where it currently does. This would help with drills, rolls, planters, ploughs etc.... Wouldn't' be good for spreaders as they throw too far back, or combines obviously. In fact with ploughing IRL when I had completed the centre work, I would often go and "pull the corners out" first in each direction and then start my headland runs, this would mean the soil under my tractor was also ploughed in the corner 

![Pulling Corners Out](https://github.com/Courseplay/CourseGenerator/assets/122838698/c1c2863c-8f5f-4553-b020-d8908f069e72)

![IMG_1348](https://github.com/Courseplay/CourseGenerator/assets/122838698/37867852-b101-48a0-9c6a-1d16695a91a4)
![IMG_1350](https://github.com/Courseplay/CourseGenerator/assets/122838698/9cb9bb6c-94bd-420d-aee6-a13114807a1e)
![IMG_1349](https://github.com/Courseplay/CourseGenerator/assets/122838698/c1f66518-92b1-4e9a-99c9-6d4da1da8123)

**Comment by coatsy35** [comment]
> Ok, I think I get it now, it definitely makes sense for implements that are long but relatively narrow so the tractor + implement is longer than the working width.
> 
> Moving to the Courseplay repo as this is a turn maneuver change, not a generator issue.

Yes that's it.

**Comment by Tensuko** [comment]
Uhm, we kinda of already do this, as long as the work width is bigger then the lenght of the vehicle.

**Comment by coatsy35** [comment]
> Uhm, we kind of do this already, as long as the work width is bigger then the length of the vehicle.

Yes it does. But a lot of implements are longer than the width unfortunately, I see it with the Kuhn Merge Max and other large trailed windrowers,, the Krone Big M Mower, most trailed mowers like the Kongskilde GXT13005P comes to mind and KV 4140L, large square balers, most ploughs, and root harvesters as you already know. 

Perhaps something can be done in the future.

**Comment by Tensuko** [comment]
Can you make a drawing with arrows how the turn should look like?

**Comment by pvaiko** [comment]
He did already, explained to me also there's a video. I know what to do.
